core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -74,6 +82,40 @@ oppia.factory('responsesService', [
           editorContextService.getActiveStateName(),
           angular.copy(newAnswerGroups));
 
+        // To check if the solution is valid once a rule has been changed or
+        // added.
+        if (
```

Comment:
Format this as:

    var currentInteractionId = stateInteractionIdService.savedMemento;
    var interactionCanHaveSolution = (
      currentInteractionId &&
      INTERACTION_SPECS[currentInteractionId].can_have_solution);
    var solutionExists = (
      stateSolutionService.savedMemento !== null &&
      stateSolutionService.savedMemento.correctAnswer !== null);

    if (interactionCanHaveSolution && solutionExists) {
      ...
    }

This gets rid of one if clause and makes the logic easier to follow.